src/main/java/core/basesyntax/service/impl/ShopServiceImpl.java

```diff
@@ -0,0 +1,19 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.ShopService;
+import core.basesyntax.strategy.OperationStrategy;
+import java.util.List;
+
+public class ShopServiceImpl implements ShopService {
+    private final OperationStrategy strategy;
+
+    public ShopServiceImpl(OperationStrategy strategy) {
+        this.strategy = strategy;
+    }
+
+    @Override
+    public void process(List<FruitTransaction> transactions) {
+        transactions.forEach(t -> strategy.get(t.getOperation()).applyOperation(t));
```

Comment:
While the lambda expression within `forEach` is concise, it's important to consider error handling. If `strategy.get(t.getOperation())` returns `null`, it will result in a `NullPointerException` when `applyOperation(t)` is called. It would be better to include a check for null and throw a custom exception with a clear message if no strategy is found for the given operation.